piston/configuration/config_loader.py

```diff
@@ -0,0 +1,94 @@
+import os
+import platform
+from typing import Optional, Union
+
+import yaml
+from piston.configuration.config_fixer import fix_config
+from piston.utilities.constants import Configuration, console
+
+
+class ConfigLoader:
+    """Loads yaml config files to customize piston-cli."""
+
+    def __init__(self, paths: Optional[Union[str, tuple]]):
+        if paths:
+            self.paths = paths
+        elif platform.system() in Configuration.configuration_paths:
+            self.paths = Configuration.configuration_paths[platform.system()]
+        self.config = {}
+
+    def _load_yaml(self) -> None:
+        """Loads the keys and values from a yaml file."""
+        expandedpath = os.path.abspath(os.path.expandvars(self.paths))
+
+        console.print(f"[green]Loading config:[/green] {expandedpath}")
+
+        with open(expandedpath) as loaded_config:
+            loaded_config = yaml.load(loaded_config, Loader=yaml.FullLoader)
+
+        for key, value in loaded_config.items():
+            if key in Configuration.default_configuration:
+                self.config[key] = value
+                console.print(f"[green]- Loaded {key}(s): {value}[/green]")
+            else:
+                console.print(
+                    f"[red]- Skipped {key}: {value} -- not a configurable value[/red]"
+                )
+
+        for key, value in Configuration.default_configuration.items():
+            if key not in self.config:
+                self.config[key] = value
+                console.print(
+                    f"[green]- Loaded default {key}: {value} -- not specified"
+                )
+
+    def load_config(self) -> dict:
+        """Loads the configuration file."""
+        path_exists = False
+
+        if type(self.paths) is str:
+            if os.path.isfile(self.paths):
+                path_exists = True
+        elif type(self.paths) is tuple:
+            for path in self.paths:
+                if os.path.isfile(path):
+                    path_exists = True
+                    self.paths = path
+                    break
+
+        if path_exists:
+            console.print(
+                "[blue]One or more configuration files was found to exist. "
```

Comment:
```suggestion
                "[blue]One or more configuration files were found to exist. "
```
`was` doesn't agree with the subject.